pytorch_lightning/accelerators/mps.py

```diff
@@ -0,0 +1,93 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Dict, List, Optional, Union
+
+import torch
+
+from pytorch_lightning.accelerators.accelerator import Accelerator
+from pytorch_lightning.utilities import device_parser
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _MPS_AVAILABLE, _PSUTIL_AVAILABLE
+from pytorch_lightning.utilities.types import _DEVICE
+
+
+class MPSAccelerator(Accelerator):
+    """Accelerator for Metal Apple M1 GPU devices."""
+
+    def setup_environment(self, root_device: torch.device) -> None:
+        """
+        Raises:
+            MisconfigurationException:
+                If the selected device is not MPS.
+        """
+        super().setup_environment(root_device)
+        if root_device.type != "mps":
+            raise MisconfigurationException(f"Device should be MPS, got {root_device} instead.")
+
+    def get_device_stats(self, device: _DEVICE) -> Dict[str, Any]:
+        """Get M1 (cpu + gpu) stats from ``psutil`` package."""
+        return get_device_stats()
+
+    @staticmethod
+    def parse_devices(devices: Union[int, str, List[int]]) -> Optional[List[int]]:
+        """Accelerator device parsing logic."""
+        parsed_devices = device_parser.parse_gpu_ids(devices, include_mps=True)
+        return parsed_devices
+
+    @staticmethod
+    def get_parallel_devices(devices: Union[int, str, List[int]]) -> List[torch.device]:
+        """Gets parallel devices for the Accelerator."""
+        parsed_devices = device_parser.parse_gpu_ids(devices, include_mps=True)
+        assert parsed_devices is not None
+
+        return [torch.device("mps")] * len(parsed_devices)
```

Comment:
But you mentioned in the docs, that it supports one device at a time?

Reply:
yes, and `parse_gpu_ids for mps always only returns a single device. This is implemented that way, that we only need to change the device parsing logic if there will ever be a machine with multiple gpus (which I assume is not that unlikely given the still missing pro setup fro the mac line).